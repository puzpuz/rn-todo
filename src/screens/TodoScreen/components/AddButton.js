import React, { useEffect, useState } from "react";
import { Text, TouchableOpacity, View } from "react-native";

import styles from "./styles/addButtonStyles";

const AddButton = ({ onAddTodo = () => {}, updateItem = undefined }) => {
  const [isUpdate, setIsUpdate] = useState(false);

  useEffect(() => {
    if (!updateItem) setIsUpdate(false);
    else {
      setIsUpdate(true);
    }
  }, [updateItem]);

  return (
    <TouchableOpacity onPress={onAddTodo}>
      <View style={styles.buttonContainer}>
        <Text style={styles.text}>{isUpdate ? "UPDATE" : "ADD"}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default AddButton;
