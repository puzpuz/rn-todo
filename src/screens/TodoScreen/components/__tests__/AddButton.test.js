import renderer from "react-test-renderer";
import AddButton from "../AddButton";
it("renders correctly", () => {
  const tree = renderer.create(<AddButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
