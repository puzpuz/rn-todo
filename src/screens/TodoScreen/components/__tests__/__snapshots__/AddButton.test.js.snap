// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "blue",
        "borderRadius": 24,
        "padding": 24,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      ADD
    </Text>
  </View>
</View>
`;
